### PR TITLE
config(amazonq): patch language configuration not added correctly #5663

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Problem

patch #5663 

Due to this line of code https://github.com/aws/aws-toolkit-vscode/blob/master/packages/core/src/codewhisperer/util/runtimeLanguageContext.ts#L156-L173

when users make inline completion call with new languages, logs will show
`2024-10-24 08:52:55.501 [error] codewhisperer: unknown runtime language dart`,  however it doesn't have functional impact because it will fallback to original language string (which is the reason why it doesn't fail test cases) and it's only affecting logs.

test has been added https://github.com/aws/aws-toolkit-vscode/blob/master/packages/amazonq/test/unit/codewhisperer/util/runtimeLanguageContext.test.ts#L184-L222

## Solution

---

License: I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by leigaol — APPROVED** [review]

**Review by zixlin7 — APPROVED** [review]

**Comment by leigaol** [comment]
Typo in description

File: packages/core/src/codewhisperer/util/runtimeLanguageContext.ts [hunk 1]
```diff
@@ -21,11 +21,17 @@ const runtimeLanguageSet: ReadonlySet<RuntimeLanguage> = new Set([
     'kotlin',
     'php',
     'python',
+    'powershell',
```
Comment (Will-ShaoHua) [inline packages/core/src/codewhisperer/util/runtimeLanguageContext.ts:24]: verified, equal to jb
`const val ID = "powershell"`

File: packages/core/src/codewhisperer/util/runtimeLanguageContext.ts [hunk 2]
```diff
@@ -21,11 +21,17 @@ const runtimeLanguageSet: ReadonlySet<RuntimeLanguage> = new Set([
...
+    'r',
+    'dart',
```
Comment (Will-ShaoHua) [inline packages/core/src/codewhisperer/util/runtimeLanguageContext.ts:26]: `const val ID = "dart"`

File: packages/core/src/codewhisperer/util/runtimeLanguageContext.ts [hunk 3]
```diff
@@ -21,11 +21,17 @@ const runtimeLanguageSet: ReadonlySet<RuntimeLanguage> = new Set([
...
     'ruby',
     'rust',
     'scala',
     'shell',
     'sql',
+    'swift',
```
Comment (Will-ShaoHua) [inline packages/core/src/codewhisperer/util/runtimeLanguageContext.ts:32]: `const val ID = "swift"`

File: packages/core/src/codewhisperer/util/runtimeLanguageContext.ts [hunk 4]
```diff
@@ -21,11 +21,17 @@ const runtimeLanguageSet: ReadonlySet<RuntimeLanguage> = new Set([
...
+    'lua',
```
Comment (Will-ShaoHua) [inline packages/core/src/codewhisperer/util/runtimeLanguageContext.ts:33]: `const val ID = "lua"`

File: packages/core/src/codewhisperer/util/runtimeLanguageContext.ts [hunk 5]
```diff
@@ -21,11 +21,17 @@ const runtimeLanguageSet: ReadonlySet<RuntimeLanguage> = new Set([
...
+    'vue',
```
Comment (Will-ShaoHua) [inline packages/core/src/codewhisperer/util/runtimeLanguageContext.ts:34]: `const val ID = "vue"`

**Review by andrewyuq — APPROVED** [review]

**Review by jpinkney-aws — APPROVED** [review]

**Pull request merged**